core/src/blocktree.rs

```diff
@@ -113,7 +113,7 @@ impl Blocktree {
         // Open the database
         let db = Database::open(&ledger_path)?;
 
-        let batch_processor = Arc::new(RwLock::new(db.batch_processor()));
+        let batch_processor = unsafe { Arc::new(RwLock::new(db.batch_processor())) };
```

Comment:
dunno why this is here  please explain

Reply:
This unsafe is just a warning that db.batch_processor() shouldn't be used unless the caller is sure that their batch operation won't conflict with any other ongoing batch API's in blocktree.

Some context:

db.batch_processor() returns a BatchProcessor object that can be used to simultaneously modify many different columns. It's used by most write operations in blocktree, in API's like `insert_data_blob`.

The safe way to get a BatchProcessor is by grabbing the RwLock for around the `blocktree.batch_processor` object, which is done by most write operations in blocktree, thanks to @mark-solana's changes. This ensures any batch write API's like like `insert_data_blob` that potentially touch overlapping column families are synchronized.

However, there are some operations like `set_root` that touch a column family (the root column family) which is not modified by other blocktree API's. If we want to use a batch operation here,
in order to avoid grabbing the shared RwLock around the `blocktree.batch_processor` object and competing with other blocktree operations, we directly make another batch_processor by calling the underlying `db.batch_processor()` function.

The `unsafe` is just a warning to people in the future to ensure that they know what they're doing when they directly call the `db.batch_processor()` to create another batch processor instead of using the one in blocktree.

@mark-solana, maybe a cleaner way to go about this eventually is to make different locks for each column family. Then we initialize a `BatchProcessor` with some subset of these locks, and only allow insertions into  this `BatchProcessor` if the insertions are into column families for which the `BatchProcessor` was passed locks during initialization

Reply:
@rob-solana, does that clarify the intention?